src/Admin.ts

```diff
@@ -120,6 +123,183 @@ export default class Admin {
+                } catch (e) {
+                    console.log(e)
+                    // invalid user
+                }
+                continue;
+            }
+            break;
+        };
+
+        let argumentsWithoutUsers = args.slice(streak);
+
+        let reason = this.handleReason(argumentsWithoutUsers.join(" "));
+
+        return {
+            members,
+            reason
+        }
+
+    }
+
+    public handleReason(reason: string) {
+        if (reason.length <= 0) return ""
+
+        if (reason.charAt(0) === ".") {
+            const noteReason = this.notes.fetchInfo(reason.split(" ")[0].substring(1), true)
+            if (noteReason !== null) {
+                const rest = reason.split(" ").slice(1).join(" ")
+                return noteReason.message + ((rest.length > 0) ? "\n" + rest : "");
+            }
+        }
+
+        return reason;
+    }
+
+
+    public getDMEmbed(reason: string, action: "banned" | "kicked", message: Discord.Message) {
+
+        function capitalizeFirstLetter(string: string) {
+            return string.charAt(0).toUpperCase() + string.slice(1);
+        }
+
+        const iconUrl = message.guild?.iconURL() || "https://upload.wikimedia.org/wikipedia/commons/1/19/Stop2.png";
+        const serverName = message.guild?.name || "Botty McBotface"
+
+        return new Discord.MessageEmbed()
+            .setTitle(`${capitalizeFirstLetter(action)} - ${serverName}`)
+            .setColor(0xff0000)
+            .setThumbnail(iconUrl)
+            .setDescription(`You were ${action} from the ${serverName}. \n\n**Reason**\n${reason}`)
+    }
+
+    private shorten(string: string) {
+        if (string.length > 400) {
+            return string.slice(0, 400) + "..."
+        }
+        return string;
+    }
+
+    public async onBan(message: Discord.Message, isAdmin: boolean, command: string, args: string[], separators: string[]) {
+        if (!isAdmin || args.length === 0) {
+            return;
+        }
+
+        const { reason, members } = await this.handleMultiUserArguments(message, args);
+
+        if (members.length === 0) { return; }
+        let removed = 0;
+
+        for (let i = 0; i < members.length; i++) {
+            const member = members[i];
+            let note = "The user was not informed as no ban message was given. ";
+
+            if (!member.bannable) {
+                this.replySecretMessage(message, `${member} is not bannable. `);
+                continue;
+            }
+
+            if (reason.length > 0) {
+                note = "The user received their ban message. ";
+                await member.send({ embed: this.getDMEmbed(reason, "banned", message) }).catch(e => {
+                    note = "The user did not receive their ban message. "
+                });
+            }
+
+            let catched = false;
+            member.ban({ reason: (reason.length > 0 ? this.shorten(reason) : "No reason") + " -" + message.author.username }).catch(e => {
```

Comment:
```js
.then(() => {
if (reason.length > 0) {
    this.replySecretMessage(message, `${member} was banned by ${message.author.username} due to "${reason}". ${note}`);
...

}).catch(e => {
...
```
Would probably look better

Reply:
I refactored the design a little bit earlier, that's why I can't 1:1 use this code. 
I hope the new commit still matches your idea (LL 259 - 267)